lib/twiglet/logger.rb

```diff
@@ -40,12 +40,14 @@ def warning(message)
 
     def error(message, error = nil)
       if error
-        message = message.merge({
-                                  'error': {
-                                    'message': error.message,
-                                    'stack_trace': error.backtrace.join("\n")
-                                  }
-                                })
+        message = message.merge(
+          {
+            'error': {
+              'message': error.message,
+              'stack_trace': (error.backtrace.join("\n") if error.backtrace)
```

Comment:
```suggestion
              'stack_trace': error.backtrace ? error.backtrace.join("\n") : ""
```
and ditch the compact. Also, I'd be suspicious of any error that doesn't have a stacktrace...

Reply:
Actually a better approach might be to add a .addStackTrace method to our Hash class.

Reply:
The problem with the suggested is that it has an empty stack_trace value. The compact removes this value so it doesn't exist in the final JSON.

Reply:
I think you're misinterpreting what I've said, I'm suggesting that we create a .addStackTrace method that we call, that will add the attribute if the stack trace is present and do nothing if it is absent.

Reply:
Aha I like that!

Reply:
Done!